daffodil/parser.py

```diff
@@ -390,10 +393,6 @@ def make_predicate(self, tokens, parent=None):
                 test = self.delegate.mk_test(tokens.pop(0).content)
                 val = self._read_val(tokens)
 
-                is_datapoint_test = getattr(test, "is_datapoint_test", False)
```

Comment:
moved this to a different part of the parser because the `is_datapoint_test` flag is an implementation detail of _some_ delegates and this breaks the encapsulation of delegates